CMakeLists.txt

```diff
@@ -25,13 +25,19 @@ endif()
 add_library(tiny-process-library ${process_source_files})
 
 target_link_libraries(tiny-process-library ${CMAKE_THREAD_LIBS_INIT})
+target_include_directories(tiny-process-library PUBLIC ${CMAKE_CURRENT_SOURCE_DIR})
 
-add_executable(examples examples.cpp)
-target_link_libraries(examples tiny-process-library)
+option(TPL_BUILD_EXAMPLES OFF)
```

Comment:
Thank you, looks good, but do you know if there is a common way now to for instance build examples and tests if `CMAKE_CURRENT_SOURCE_DIR` is equal to `PROJECT_SOURCE_DIR`, that is if tiny-process-library is the main/top-level project.

Reply:
CMAKE_PROJECT_NAME variable will have the value of the top level project so f.i.

    if ("${CMAKE_PROJECT_NAME}" STREQUAL "tiny-process-library")
        set(TPL_STANDALONE ON)
    endif()

Then you can use TPL_STANDALONE for stuff that you only be included then.

Reply:
Or you can use:

    if ("${CMAKE_SOURCE_DIR}" STREQUAL "${CMAKE_CURRENT_SOURCE_DIR}")
        ....
    endif()

to accomplish the same thing.

Reply:
I was thinking that it might be a bit unorthodox to do it in this way, but the second option you list looks ok I think. I'll look into this a bit more and see if there is a common way to do this.

Reply:
I agree. The second option is "fool proof", as project names might not be unique within the source tree. Although I hope no one would use tiny-process-library as project name when using tiny-process-library ;)

Reply:
This looks promising: https://stackoverflow.com/a/25217937/255635